src/cdk/tree/tree.ts

```diff
@@ -414,6 +480,72 @@ export class CdkTree<T, K = T> implements AfterContentChecked, CollectionViewer,
   _getChildrenAccessor() {
     return this.treeControl?.getChildren ?? this.childrenAccessor;
   }
+
+  private _getAllDescendants(): Observable<T[]> {
+    return merge(...(this._dataNodes?.map(dataNode => this._getDescendants(dataNode)) ?? []));
+  }
+
+  private _getDescendants(dataNode: T): Observable<T[]> {
```

Comment:
It's not clear to me why these getters are returning observables. It seems to convert realtime results to an observable that never updates. Could these be synchronous functions?